include/fmt/chrono.h

```diff
@@ -961,6 +961,26 @@ OutputIt format_duration_unit(OutputIt out) {
   return out;
 }
 
+struct get_locale {
+  explicit get_locale(bool localized, locale_ref loc_ref)
+      : has_locale{localized} {
+    if (localized) locale = loc_ref.template get<std::locale>();
+  }
+  ~get_locale() {
+    if (has_locale) locale.~locale();
+  }
+  operator const std::locale&() const {
+    return has_locale ? locale : get_classic_locale();
+  }
+
+ private:
+  union {
+    monostate dummy{};
```

Comment:
``monostate``  is C++17 only

Reply:
`{}` seems unnecessary, please remove.

Reply:
Strictly speaking this is unnecessary.

But without `monospace` default initialized, this is kinda scary because you have a generally unintialized member, and invariants become kinda hard to follow.

All this is OK for experts, but I imagine that this is harder to follow for non-experts.
`monostate` has negligible overhead, but helps to figure out what's happening more easily.
That's why I added `monostate` member.

Reply:
`monostate` is fine but I don't think you need to initialize it explicitly at least because it has no state =).

Reply:
If you never initialize `monostate dummy`, than there is no point in having it at all.

IMHO, initializing `monostate dummy` by default gives some understanding and peace of mind to the person trying to understand the code.

Initializing it by default says that the lifetime of the `dummy` object is started, and you need to explicitly do something to change it.

In this case when `locale` is initialized using placement `new`, it explicitly says that the lifetime of `dummy` ends and another object starts to occupy the storage simultaneously starting its lifetime.

If there was no `monostate dummy` it would work perfectly correct, but then a person trying to understand the code must keep in mind that the union is generally _uninitialized_.